### PR TITLE
Fix commit attribution configuration

### LINKED ISSUE #919: Openclaude default commit co author on non claude models
## Summary

What is broken?
if you tell openclaude to commit for you. the commit message is usually appended with the model used but if it is a non claude model it just defaults to opus 4.6 and also injects a email default email address in the message as well.

## Steps to Reproduce

1. use model other than claude models
2. ask openclaude to generate commit
3. commits with default claude info

## Expected Behavior

What should have happened?
there should be a /commit-message command where you can configure this or turn it off.

## Actual Behavior

What happened instead?

## Environment

- OpenClaude version:
- OS:
- Terminal:
- Provider:
- Model:

## Logs / Screenshots

Paste the exact error output or attach screenshots if useful.

## Additional Context

Anything else maintainers should know?

### PR DESCRIPTION
**Pull request by techbrewboss**
## Summary
- add /commit-message to view, disable, reset, or customize commit attribution
- avoid labeling non-Claude provider commit co-authors as Claude Opus 4.6
- document the CLI command alongside the existing attribution setting

## Test plan
- [x] bun test src/commands/commit-message/commit-message.test.ts src/utils/attribution.test.ts
- [x] git diff --check

Fixes #919

### CODE REVIEW

File: src/commands/commit-message/commit-message.test.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { describe, expect, it } from 'bun:test'
+import { formatCoAuthorTrailer, parseCoAuthor } from './commit-message.js'
+
+describe('commit-message command helpers', () => {
+  it('parses quoted co-author names with a plain email', () => {
+    expect(parseCoAuthor('"GPT 5.5" noreply@openclaude.dev')).toEqual({
+      name: 'GPT 5.5',
+      email: 'noreply@openclaude.dev',
```
Comment (kevincodex1) [inline src/commands/commit-message/commit-message.test.ts:13]: lets use openclaude@gitlawb.com for now cause we dont have the .dev domain yet

Comment (techbrewboss) [reply src/commands/commit-message/commit-message.test.ts:13]: On it!

Comment (techbrewboss) [reply src/commands/commit-message/commit-message.test.ts:13]: addressed with commit [7749b38]

**Review by gnanam1990 — CHANGES_REQUESTED** [review]
Thanks for picking this up — the /commit-message command and the non-Claude provider attribution fix are exactly right, and the tests are appreciated.

One change needed before this can land: getDefaultCommitCoAuthorEmail still returns 'noreply@anthropic.com' for the firstParty branch. Since this fork is rebranding away from Anthropic, hardcoding that address (even on the firstParty path) leaks an Anthropic fingerprint we're trying to remove. Could you switch the default to openclaude@gitlawb.com for all providers, or drop the firstParty special case so it falls through to the OpenClaude default?

Everything else looks good — once that's updated I'm happy to approve.

**Comment by techbrewboss** [comment]
@gnanam1990 addressed in 005d936: removed the firstParty email special case so default commit co-author attribution now uses openclaude@gitlawb.com for all providers. Also updated the focused test to cover both openai and firstParty inputs.

**Review by gnanam1990 — DISMISSED** [review]
Thanks for the quick turnaround — the new getDefaultCommitCoAuthorEmail() returns the OpenClaude email for all providers (no firstParty special case), and the test asserting both openai and firstParty get the same email locks it in. Anthropic email leak is fully resolved.

LGTM.

File: src/utils/attribution.ts [hunk 2]
```diff
@@ -39,6 +39,58 @@ export type AttributionTexts = {
...
+  const isKnownPublicModel = getPublicModelDisplayName(model) !== null
+  const normalizedModel = model.toLowerCase()
+  const isClaudeProvider =
+    apiProvider === 'firstParty' ||
+    apiProvider === 'bedrock' ||
+    apiProvider === 'vertex' ||
+    apiProvider === 'foundry' ||
+    normalizedModel.includes('claude')
+
+  if (isInternalRepo || (isClaudeProvider && isKnownPublicModel)) {
```
[the remark this answers is not in the document]
Comment (techbrewboss) [reply src/utils/attribution.ts]: Addressed in 8471d5c. The internal-repo exception now only applies when `isClaudeProvider` is true, so non-Claude providers keep the OpenClaude model attribution path even in internal repos. Added a regression test for an internal OpenAI model.

File: src/utils/attribution.ts [hunk 3]
```diff
@@ -39,6 +39,58 @@ export type AttributionTexts = {
...
+    .replace(/\s+/g, ' ')
+    .trim()
+}
+
+function formatClaudeCoAuthorName(model: string): string {
+  const publicName = getPublicModelDisplayName(model)
+  if (!publicName) {
+    return getPublicModelName(model)
+  }
+  return publicName.startsWith('Claude ') ? publicName : `Claude ${publicName}`
```
[the remark this answers is not in the document]
Comment (techbrewboss) [reply src/utils/attribution.ts]: Addressed in 8471d5c. `formatClaudeCoAuthorName()` now sanitizes both the unknown-model fallback and public display-name path before returning a trailer name. Added a regression test for CR/LF and angle bracket sanitization.

File: src/utils/attribution.ts [hunk 4]
```diff
@@ -39,6 +39,58 @@ export type AttributionTexts = {
...
+    normalizedModel.includes('claude')
+
+  if (isInternalRepo || (isClaudeProvider && isKnownPublicModel)) {
+    return formatClaudeCoAuthorName(model)
+  }
+
+  // Unknown first-party models may be unreleased Claude codenames, so keep the
+  // historical public fallback. OpenAI-compatible providers should identify the
+  // actual configured model instead of claiming Claude Opus.
+  if (apiProvider === 'firstParty') {
```
[the remark this answers is not in the document]
Comment (techbrewboss) [reply src/utils/attribution.ts:87]: Addressed in 8471d5c. Added the `@[MODEL LAUNCH]` marker next to the `Claude Opus 4.6` fallback so future model updates have the same maintenance breadcrumb as the rest of the model launch code.

**Review by gnanam1990 — APPROVED** [review]
Re-approving after 8471d5c. The new commit cleanly addresses all three Copilot findings:

- Internal-repo Claude formatting now properly gated behind isClaudeProvider so OpenAI/Gemini models in internal repos don't get mislabeled as Claude
- formatClaudeCoAuthorName sanitizes both the unknown-model fallback and public display path (CR/LF + angle brackets stripped)
- @[MODEL LAUNCH] breadcrumb next to the 'Claude Opus 4.6' hardcoded fallback matches the pattern used elsewhere

Regression tests cover the new edge cases. Email leak is still resolved.

LGTM.

**Review by kevincodex1 — APPROVED** [review]
looks good to me
